common/core/overlaylayer.cpp

```diff
@@ -112,6 +112,29 @@ void OverlayLayer::ValidatePreviousFrameState(const OverlayLayer& rhs) {
   }
 }
 
+void OverlayLayer::SetSurfaceDamage(const HwcRegion& surface_damage,
+                                    const OverlayLayer& rhs) {
+  // check the surafce damage if its 0 for all layers then skip validate
+  // and commit
```

Comment:
please remove this commit from the above comments

Apart from that Looks Good for me